### ISSUE TITLE
Expo Go does not load on Android 5.1 because of reanimated

### ISSUE THREAD

**Issue by byCedric** [expo/expo#15822]
### Summary

Expo Go `2.23.2` shows a white screen after the splash screen on Android 5.1. The logcat shows and issue related to Reanimated:

```
I/ReactNativeJS( 4592): Running "main" with {"initialProps":{"exp":{}},"rootTag":1}
W/System.err( 4592): java.lang.reflect.InvocationTargetException
W/System.err( 4592):    at java.lang.reflect.Method.invoke(Native Method)
W/System.err( 4592):    at java.lang.reflect.Method.invoke(Method.java:372)
W/System.err( 4592):    at com.facebook.react.bridge.DefaultNativeModuleCallExceptionHandler.handleException(DefaultNativeModuleCallExceptionHandler.java:4)
W/System.err( 4592):    at com.facebook.react.devsupport.DisabledDevSupportManager.handleException(DisabledDevSupportManager.java:1)
W/System.err( 4592):    at com.facebook.react.bridge.ReactContext.handleException(ReactContext.java:4)
W/System.err( 4592):    at com.facebook.react.uimanager.GuardedFrameCallback.doFrame(GuardedFrameCallback.java:2)
W/System.err( 4592):    at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:7)
W/System.err( 4592):    at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame(ChoreographerCompat.java:1)
W/System.err( 4592):    at android.view.Choreographer$CallbackRecord.run(Choreographer.java:765)
W/System.err( 4592):    at android.view.Choreographer.doCallbacks(Choreographer.java:580)
W/System.err( 4592):    at android.view.Choreographer.doFrame(Choreographer.java:549)
W/System.err( 4592):    at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:753)
W/System.err( 4592):    at android.os.Handler.handleCallback(Handler.java:739)
W/System.err( 4592):    at android.os.Handler.dispatchMessage(Handler.java:95)
W/System.err( 4592):    at android.os.Looper.loop(Looper.java:135)
W/System.err( 4592):    at android.app.ActivityThread.main(ActivityThread.java:5254)
W/System.err( 4592):    at java.lang.reflect.Method.invoke(Native Method)
W/System.err( 4592):    at java.lang.reflect.Method.invoke(Method.java:372)
W/System.err( 4592):    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
W/System.err( 4592):    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
W/System.err( 4592): Caused by: com.facebook.react.common.JavascriptException: Attempt to invoke virtual method 'boolean versioned.host.exp.exponent.modules.api.reanimated.layoutReanimation.ReaLayoutAnimator.isLayoutAnimationEnabled()' on a null object reference
W/System.err( 4592):    at host.exp.exponent.ReactNativeStaticHelpers.handleReactNativeError(ReactNativeStaticHelpers.kt:8)
W/System.err( 4592):    ... 20 more
W/ReactNativeJS( 4592): Clipboard has been extracted from react-native core and will be removed in a future release. It can now be installed and imported from '@react-native-community/clipboard' instead of 'react-native'. See https://github.com/react-native-clipboard/clipboard
I/WebViewFactory( 4592): Loading com.android.webview version 39 (6695563-x86) (code 300007)
I/LibraryLoader( 4592): Time to load native libraries: 5 ms (timestamps 1076-1081)
I/LibraryLoader( 4592): Expected native library version number "",actual native library version number ""
```

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

44

### Environment

```
  Expo CLI 5.0.4 environment info:
    System:
      OS: Windows 10 10.0.22000
    Binaries:
      Node: 16.11.1 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.17 - C:\Program Files\nodejs\yarn.CMD
      npm: 8.1.0 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version     2020.3.0.0 AI-203.7717.56.2031.7784292
    Expo Workflow: managed
```

### Reproducible demo

- Create Android 5.1 (API 22) emulator
- Modify the emulator default memory to something bigger (e.g. 8gb for internal and SD memory)
    > Else it might fail with insufficient storage errors.
- `$ expo client:install:android`
- `$ adb logcat`
- Open Expo Go

**Comment by Douglas-Cezaro** [comment]
I'm having the same problem, is there anything I can do to fix this?

**Comment by brentvatne** [comment]
related pr: https://github.com/software-mansion/react-native-reanimated/pull/2847

**Comment by brentvatne** [comment]
@Glazzes - on android 5.1? then you're in the right place! this is indeed a bug as mentioned in this issue :)

**Comment by superyarik** [comment]
@esadkrs facing this issue with `react-native-reanimated` `2.1.0` version

**Comment by raphaelmerx** [comment]
The fix was released in [react-native-reanimated@2.4.0](https://github.com/software-mansion/react-native-reanimated/releases/tag/2.4.0), however expo 44 is not marked as compatible with that version of `react-native-reanimated`, resulting in a warning when installing 2.4:
```
Some dependencies are incompatible with the installed expo package version:
 - react-native-reanimated - expected version: ~2.3.1 - actual version installed: 2.4.1
Your project may not work correctly until you install the correct versions of the packages.
```
